nmigen/lib/fifo.py

```diff
@@ -279,7 +279,10 @@ class AsyncFIFO(Elaboratable, FIFOInterface):
         Always set.
     """.strip(),
     r_data_valid="Valid if ``r_rdy`` is asserted.",
-    r_attributes="",
+    r_attributes="""
+    r_rst : out
+        Asserted when the FIFO is reset by the write-domain reset.
```

Comment:
Asserted *when*? That implies to me that it's something like a 1-clock pulse. In reality it is asserted *while*, right?

Reply:
I wrote _while_ initially, but the write domain reset pulse can technically be shorter than a read domain clock cycle, so I felt like it was inaccurate. Probably needlessly pedantic though - you're right that _when_ implies a single pulse (which we could do, i guess).

Reply:
I think holding it while the reset is asserted is a lot more useful, since you can go to a pulse from that but not the other way around.

Regaring pedantry, I think pedantry around CDC is pretty much a necessity. I think it would be very useful if we wrote that it is asserted "while the write-domain reset is, but no less than one read-domain clock cycle" or something like that.

Reply:
(No less than two in practice, but do we want to commit to that?)

Reply:
OK, I agree with that. "Asserted while the FIFO is being reset by the write-domain reset (for at least one read-domain clock cycle)"